cli/popper/gha.py

```diff
@@ -32,9 +32,9 @@
 class WorkflowRunner(object):
     """A GHA workflow runner."""
 
-    def __init__(self, workflow):
+    def __init__(self, workflow, substitutions=None, allow_loose=False):
         self.wf = workflow
-        self.wf.parse()
+        self.wf.parse(substitutions, allow_loose)
```

Comment:
if we make these arguments be part of the `Workflow` constructor, then we can skip passing them to the `WorkflowRunner` constructor and we'd remove them from this function altogether

Reply:
Fixed.